### PR TITLE
Smingolelliwebcontroller

### PR DESCRIPTION
**Pull request by smingolelli**

### CODE REVIEW

File: JavaProject/Game/src/equationscontroller/Equations_Controller.java [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+		case 0:
+			return this.moveToForbidden(index);
+		case 1:
+			return this.moveToRequired(index);
+		case 2:
+			return this.moveToAllowed(index);
+
+		}
+
+		// If we're lucky, this shouldn't occur.
```
Comment (DerrekWoodworth) [inline JavaProject/Game/src/equationscontroller/Equations_Controller.java]: lol

File: controller/Equations_Controller.java [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+	 * @param playerInput
+	 */
+	public boolean challengeImpossible(String playerInput) {
+		Algebra a = new Algebra(playerInput, myGoal);
+		Solver s = new Solver(a);
+		if (s.validEquation(playerInput) == false)
+			return false;
+
+		// If this is true, then the challenge was not impossible.
+		if (s.checkAnswer(playerInput) == true)
```
Comment (DerrekWoodworth) [inline controller/Equations_Controller.java]: You don't need the `== true`. checkAnswer returns a boolean for the if statement to use

File: controller/Equations_Controller.java [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+	/**
+	 * Check if the challenge is impossible. The function will return false if the
+	 * equation is not valid.
+	 * 
+	 * @param playerInput
+	 */
+	public boolean challengeImpossible(String playerInput) {
+		Algebra a = new Algebra(playerInput, myGoal);
+		Solver s = new Solver(a);
+		if (s.validEquation(playerInput) == false)
```
Comment (DerrekWoodworth) [inline controller/Equations_Controller.java]: `if (s.validEquation(playerInput) == false)`
Can be written as
`if (!s.validEquation(playerInput))`

File: controller/Equations_Controller.java [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+	 */
+	public boolean challengeNow(String playerInput) {
+		Algebra a = new Algebra(playerInput, myGoal);
+		Solver s = new Solver(a);
+		if (s.validEquation(playerInput) == false)
+			return false;
+
+		// If this is true, then the player got the correct answer
+		if (s.checkAnswer(playerInput) == true)
+			return true;
```
Comment (DerrekWoodworth) [inline controller/Equations_Controller.java]: Same thing with the if statements here

File: controller/Equations_Controller.java (hunk 2 above)
Comment (smingolelli) [reply controller/Equations_Controller.java]: True, it has become more habit if anything. It also looks more readable to me. Same for the other if statement.

**Pull request merged**